### PR TITLE
Avoid use of strdup, strndup

### PR DESCRIPTION
**Pull request by Hello71**
* Avoid use of strndup

On old glibc, strdup and strndup are aliased to __strdup and __strndup,
which musl does not implement. This causes netty to unnecessarily fail
on musl. strndup is not the right tool here anyways, since we know the
exact size of the string to be duplicated, not the maximum size.

* Don't free packagePrefix after load_function

This allows load_function to avoid calling strdup, which is both more
efficient and also improves musl compatibility.

### CODE REVIEW

File: src/main/c/netty_jni_util.c [hunk 1]
```diff
@@ -461,7 +437,6 @@ jint netty_jni_util_JNI_OnLoad(JavaVM* vm, void* reserved, const char* libname,
 #endif /* NETTY_JNI_UTIL_BUILD_STATIC */
 
     jint ret = load_function(env, packagePrefix);
-    free(packagePrefix);
```
Comment (normanmaurer) [inline src/main/c/netty_jni_util.c:464]: why this change ?

Comment (Hello71) [reply src/main/c/netty_jni_util.c:464]: as explained in the commit descriptions, the goal is to avoid strdup in the load_function. the reason why strdup was called in the first place was because the packagePrefix is invalid after load_function returns. we can avoid this by simply passing the ownership of packagePrefix to load_function. if we free packagePrefix then it will be a UAF.

Comment (normanmaurer) [reply src/main/c/netty_jni_util.c:464]: got it... thanks!
